File: kommo_scraper.py
from playwright.sync_api import sync_playwright
import time
import os

def main():
    os.makedirs('kommo_screenshots', exist_ok=True)
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        context = browser.new_context(viewport={'width': 1280, 'height': 800})
        page = context.new_page()
        
        print("Acessando Kommo...")
        page.goto('https://marcelogeusti.kommo.com/')
        page.wait_for_load_state('networkidle')
        page.screenshot(path='kommo_screenshots/01_login_page.png')
        
        print("Preenchendo login...")
        # A página de login pode ter seletores específicos. Vamos tentar os genéricos de e-mail/senha.
        try:
            page.fill('input[type="text"], input[type="email"], input[name="USER_LOGIN"]', 'marcelogeus@gmail.com')
            page.fill('input[type="password"], input[name="USER_PASSWORD"]', 'G@usti8826')
            page.screenshot(path='kommo_screenshots/02_filled_login.png')
            
            # Clicar em login
            page.click('button[type="submit"], input[type="submit"], button:has-text("Log in"), button:has-text("Entrar")')
            print("Aguardando carregamento do dashboard...")
            page.wait_for_load_state('networkidle')
            time.sleep(5) # Extra wait for dynamic content
            page.screenshot(path='kommo_screenshots/03_dashboard.png', full_page=True)
            
            # Print URL to see where we landed
            print("URL atual:", page.url)
            
            # Navegar para as páginas
            # Pipeline/Leads
            page.goto('https://marcelogeusti.kommo.com/leads/pipeline/')
            page.wait_for_load_state('networkidle')
            time.sleep(3)
            page.screenshot(path='kommo_screenshots/04_pipeline.png', full_page=True)
            
            # Lista de Contatos
            page.goto('https://marcelogeusti.kommo.com/contacts/list/')
            page.wait_for_load_state('networkidle')
            time.sleep(3)
            page.screenshot(path='kommo_screenshots/05_contacts.png', full_page=True)

            # Analytics / Dashboard
            page.goto('https://marcelogeusti.kommo.com/dashboard/')
            page.wait_for_load_state('networkidle')
            time.sleep(3)
            page.screenshot(path='kommo_screenshots/06_analytics.png', full_page=True)

        except Exception as e:
            print(f"Erro: {e}")
            page.screenshot(path='kommo_screenshots/error.png')
            
        browser.close()

if __name__ == '__main__':
    main()

File: scripts/legacy/kommo_scraper.js
const puppeteer = require('puppeteer');
const fs = require('fs');

async function scrape() {
  if (!fs.existsSync('kommo_screenshots')){
      fs.mkdirSync('kommo_screenshots');
  }

  console.log('Iniciando Puppeteer...');
  const browser = await puppeteer.launch({ headless: true, defaultViewport: { width: 1280, height: 800 } });
  const page = await browser.newPage();

  try {
    console.log('Acessando Login...');
    await page.goto('https://marcelogeusti.kommo.com/', { waitUntil: 'networkidle2' });
    await page.screenshot({ path: 'kommo_screenshots/01_login.png' });

    console.log('Preenchendo Login...');
    // Procurar por input de login (username/email)
    const loginSelector = 'input[type="text"], input[type="email"], input[name="USER_LOGIN"]';
    const passSelector = 'input[type="password"], input[name="USER_PASSWORD"]';
    
    await page.waitForSelector(loginSelector);
    await page.type(loginSelector, 'marcelogeus@gmail.com');
    await page.type(passSelector, 'G@usti8826');
    await page.screenshot({ path: 'kommo_screenshots/02_filled.png' });

    console.log('Clicando em Entrar...');
    await page.evaluate(() => {
        const btn = document.querySelector('button[type="submit"], input[type="submit"], .js-login-submit, button.button-input');
        if (btn) btn.click();
    });

    console.log('Aguardando dashboard...');
    await page.waitForNavigation({ waitUntil: 'networkidle2', timeout: 15000 }).catch(e => console.log('Navigation timeout, continuing...'));
    await new Promise(r => setTimeout(r, 5000));
    await page.screenshot({ path: 'kommo_screenshots/03_dashboard.png', fullPage: true });

    // Pipeline
    console.log('Acessando Pipeline...');
    await page.goto('https://marcelogeusti.kommo.com/leads/pipeline/', { waitUntil: 'networkidle2' });
    await new Promise(r => setTimeout(r, 3000));
    await page.screenshot({ path: 'kommo_screenshots/04_pipeline.png', fullPage: true });

    // Clicar em um lead se houver para ver o perfil
    console.log('Tentando abrir um lead...');
    await page.evaluate(() => {
        const lead = document.querySelector('.pipeline_leads__item');
        if (lead) lead.click();
    });
    await new Promise(r => setTimeout(r, 3000));
    await page.screenshot({ path: 'kommo_screenshots/05_lead_profile.png', fullPage: true });

    // Contacts
    console.log('Acessando Contatos...');
    await page.goto('https://marcelogeusti.kommo.com/contacts/list/', { waitUntil: 'networkidle2' });
    await new Promise(r => setTimeout(r, 3000));
    await page.screenshot({ path: 'kommo_screenshots/06_contacts.png', fullPage: true });

  } catch (error) {
    console.error('Erro durante o scrape:', error);
    await page.screenshot({ path: 'kommo_screenshots/error.png' });
  } finally {
    console.log('Fechando...');
    await browser.close();
  }
}

scrape();
